test/irmin/test.ml

```diff
@@ -265,6 +266,80 @@ let test_to_string () =
   test "record" my_record_t
     { foo = 2; flag = false; letter = Delta }
     "{\"foo\":2,\"flag\":0,\"letter\":\"Delta\"}";
+
+  ()
+
+(** Test the behaviour of {!Irmin.Type.pp_ty}. *)
+let test_pp_ty () =
+  let test : type a. string -> a T.t -> string -> unit =
+   fun case_name input expected_output ->
+    let assertion =
+      Fmt.strf "Expected output of `pp_ty` for representation of `%s`" case_name
+    in
+    (Fmt.to_to_string T.pp_ty) input
+    |> Alcotest.(check string) assertion expected_output
+  in
+
+  (* Test cases for basic types *)
+  test "unit" T.unit "Prim Unit";
+  test "bool" T.bool "Prim Bool";
+  test "char" T.char "Prim Char";
+  test "int" T.int "Prim Int";
+  test "int32" T.int32 "Prim Int32";
+  test "int64" T.int64 "Prim Int64";
+  test "float" T.float "Prim Float";
+  test "bytes" T.bytes "Prim Bytes";
+  test "string" T.string "Prim String";
+
+  (* Test cases for non-algebraic combinators *)
+  test "int list" T.(list int) "List (Prim Int)";
+  test "float array" T.(array float) "Array (Prim Float)";
+  test "(unit * int)" T.(pair unit int) "Pair (Prim Unit, Prim Int)";
+  test "unit option" T.(option unit) "Option (Prim Unit)";
+  test "(int * string * bool)"
+    T.(triple int string bool)
+    "Triple (Prim Int, Prim String, Prim Bool)";
+  test "(string, bool) result" T.(result string bool) "Variant";
+
+  (* Test cases for fixed-size refinement types *)
+  test "string {size=Int}" T.(string_of `Int) "Prim String";
+  test "string {size=Int8}" T.(string_of `Int8) "Prim String:8";
+  test "bytes {size=Int16}" T.(bytes_of `Int16) "Prim Bytes:16";
+  test "bytes {size=Int32}" T.(bytes_of `Int32) "Prim Bytes:32";
+  test "array {size=Int64}" T.(array ~len:`Int64 unit) "Array:64 (Prim Unit)";
+  test "array {size=3}" T.(array ~len:(`Fixed 3) unit) "Array:<3> (Prim Unit)";
+
+  (* Test cases for algebraic combinators *)
+  test "enum" Algebraic.my_enum_t "Variant";
```

Comment:
Do we want to be a bit more precise for variants and records? I guess these printers are just for debugging anyway, but we could imagine extending them one day to become proper schema, and having a precise printer would help parsing. I agree this is clearly not very urgent to do :-)